Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1356,7 +1356,7 @@ withTempFileEx opts tmpDir template action =
 withTempDirectory :: Verbosity -> FilePath -> String -> (FilePath -> IO a) -> IO a
 withTempDirectory verbosity targetDir template f = withFrozenCallStack $
   withTempDirectoryEx verbosity defaultTempFileOptions targetDir template
-    (withLexicalCallStack f)
+    (withLexicalCallStack (\x -> f x))
```

Comment:
@RyanGlScott I'll leave this in place, yet I'm not sure whether we need (i.e. there's any benefit) from `wthFrozen/LexicalCallStack`.